### DISCUSSION TITLE
Docker and git-cliff

### DISCUSSION THREAD

**Discussion by nikoskontothanasis** [orhun/git-cliff#603] [Q&A]
Hello, 
I am not sure if i am using the tool in the correct way, but i would like your advise

i run in the root dir of my repo the command: 
$ docker run -t -v "$(pwd)":/app/ orhunp/git-cliff:latest
and return an error: 
 WARN  git_cliff > "cliff.toml" is not found, using the default configuration.
 ERROR git_cliff > Git error: `could not find repository at '/app'; class=Repository (6); code=NotFound (-3)`

In case i run this: 
$ docker run -t -v "$(pwd)":/app/ git-cliff --init
succeed 
INFO  git_cliff > Saving the configuration file to "cliff.toml"

Any recommendations on how can i use it with docker?

**Comment by orhun** [comment]
Hey!

Normally, the first command you provided should work out of the box. Also, is `cliff.toml` actually saved in the repo when you run `--init`?

Can you provide more information about your set up? I'm not sure what is going wrong there.

**Reply by nikoskontothanasis** [reply]
No it is not create the cliff.toml file in the repo 
Is it mandatory to install in my local setup the git-cliff to use it from docker command? 
Because it is not clear if it is. 
I dont have install anything, i just run the docker command in the root repo dir.

**Reply by orhun** [reply]
Yeah, you only need to run the docker command and do not install anything.

I still don't get how you are running it, can you share more information?

**Reply by nikoskontothanasis** [reply]
I clone my repo, cd to my repo and run the command docker run -t -v "$(pwd)":/app/ orhunp/git-cliff:latest --init

**Reply by orhun** [reply]
You don't have to use `--init`, normally running `docker run -t -v "$(pwd)":/app/ orhunp/git-cliff:latest` should just work out of the box.

Just to confirm, can you pull the image again? (`docker pull orhunp/git-cliff:latest`)

**Reply by nikoskontothanasis** [reply]
i see. But if i run only docker run -t -v "$(pwd)":/app/ orhunp/git-cliff:latest 
it returns me error: 
WARN git_cliff > "cliff.toml" is not found, using the default configuration.
ERROR git_cliff > Git error: could not find repository at '/app'; class=Repository (6); code=NotFound (-3)

yes i can pull the image docker pull orhunp/git-cliff:latest
Digest: sha256:4f18f90df0f5429d5609c4903205aceba231b8621e34f62ee10e3421b40b6e8c

**Reply by orhun** [reply]
Very weird.

Can you run the following command and confirm that the project files are mounted?

```sh
$ docker run -t -v "$(pwd)":/app/ --entrypoint "ls" orhunp/git-cliff:latest
```

Alternatively, you can this command to shell into the container and run `git cliff` manually:

```sh
$ docker run -t -v "$(pwd)":/app/ -it --entrypoint "sh" orhunp/git-cliff:latest

$ git-cliff
```

Can you let me know about the output?

**Reply by giornn0** [reply]
Hi, in case anyone trying to use it on powershell, this cmd worked for me:
`docker run -t -v ${pwd}:/app/ orhunp/git-cliff:latest`

On nushell I was only able to make it run with without `$(pwd)` (no idea why):
`docker run -t -v ./:/app/ orhunp/git-cliff:latest`
